### PR TITLE
Add Kado

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206626267967685

### CODE REVIEW

File: src/partners/kado.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  // message: asString,
+  data: asObject({
+    onRamps: asArray(asOnRampTx),
+    offRamps: asArray(asOffRampTx)
+  })
+})
+
+const MAX_RETRIES = 5
+const QUERY_LOOKBACK = 1000 * 60 * 60 * 24 * 30 // 30 days
+const QUERY_TIME_BLOCK_MS = QUERY_LOOKBACK
```
Comment (samholmes) [inline src/partners/kado.ts:62]: Why the extra constant?

File: src/partners/kado.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  let startTime = lastCheckedTimestamp
+
+  // No looping since all txs are returned in one call for now
+  // while (true)
+  {
+    const endTime = startTime + QUERY_TIME_BLOCK_MS
+    const now = Date.now()
+
+    const startTimeS = startTime / 1000
+    const endTimeS = endTime / 1000
```
Comment (samholmes) [inline src/partners/kado.ts:96]: how does this not give you unused variable errors?

File: src/partners/kado.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  let lastCheckedTimestamp = new Date(latestIsoDate).getTime() - QUERY_LOOKBACK
+  if (lastCheckedTimestamp < 0) lastCheckedTimestamp = 0
+
+  const ssFormatTxs: StandardTx[] = []
+  let retry = 0
+  let startTime = lastCheckedTimestamp
+
+  // No looping since all txs are returned in one call for now
+  // while (true)
+  {
```
Comment (samholmes) [inline src/partners/kado.ts:91]: Why not just make a separate commit for the while-loop and time variables which can be in a separate wip-branch?

**Review by samholmes — APPROVED** [review]
Approved, but I think cleanup by putting dead-code in a separate commit and separate wip-branch may be better then committing it now.

**Comment by paullinator** [comment]
@samholmes since this code is already deployed in develop, i'll see if they update the API in the near future and I'll just modify this PR for the new API. Otherwise if I get impatient, I'll pull out the dead code.
